lib/instrumentation/context.js

```diff
@@ -0,0 +1,97 @@
+'use strict'
+
+const crypto = require('crypto')
+
+function fromHex (hex) {
+  return Buffer.from(hex, 'hex')
+}
+
+function toHex (buffer) {
+  return buffer.toString('hex')
+}
+
+class TraceContext {
+  constructor ({ version, traceId, id, flags, parentId } = {}) {
+    // 1 byte
+    this.version = version
+    // 16 bytes
+    this.traceId = traceId
+    // 8 bytes
+    this.id = id
+    // 1 byte
+    this.flags = flags
+    // 8 bytes
+    this.parentId = parentId
+  }
+
+  static create (sampled = false) {
+    const bytes = crypto.randomBytes(24)
+    const flags = Buffer.alloc(1)
+
+    // Load sampled flag into first bit of flags
+    if (sampled) {
+      flags[0] |= 0x01
+    } else {
+      flags[0] &= ~0x01
+    }
+
+    return new TraceContext({
+      version: Buffer.alloc(1),
+      traceId: bytes.slice(0, 16),
+      id: bytes.slice(16),
+      flags: flags
+    })
+  }
+
+  static fromString (header) {
+    const [
+      version,
+      traceId,
+      id,
+      flags
+    ] = header.split('-').map(fromHex)
+
+    return new TraceContext({
+      version,
+      traceId,
+      id,
+      flags
+    })
+  }
+
+  child () {
+    const { version, traceId, id: parentId, flags } = this
+    const id = crypto.randomBytes(8)
+
+    return new TraceContext({
+      version,
+      traceId,
+      id,
+      flags,
+      parentId
+    })
+  }
+
+  toString () {
+    const parts = [
+      this.version,
+      this.traceId,
+      this.id,
+      this.flags
+    ]
+
+    return parts.map(toHex).join('-')
+  }
+
+  toJSON () {
```

Comment:
`toJSON` was originally intended just for debug purposes just because the buffer form is not really readable. I ended up using it in a couple spots internally though. Might be better to just have a private buffer and string getters though? 🤔

Reply:
I tried a new implementation with a shared buffer and string getters.